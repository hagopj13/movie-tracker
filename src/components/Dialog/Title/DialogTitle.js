// @flow
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

type Props = {
  title: string,
  onClose: () => void,
};

const useStyles = makeStyles((theme) => ({
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
}));

const DialogTitle = (props: Props) => {
  const { title, onClose, ...otherProps } = props;
  const classes = useStyles();

  return (
    <MuiDialogTitle disableTypography {...(otherProps: any)}>
      <Typography variant="h6">{title}</Typography>
      <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
        <CloseIcon />
      </IconButton>
    </MuiDialogTitle>
  );
};

export default DialogTitle;
