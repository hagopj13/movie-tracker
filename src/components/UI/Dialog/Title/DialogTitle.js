import React from 'react';
import type { Node } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

type Props = {
  children: Node,
  onClose: () => void,
};

const useStyles = makeStyles((theme) => ({
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
}));

const DialogTitle = (props: Props) => {
  const { children, onClose, ...otherProps } = props;
  const classes = useStyles();

  return (
    <MuiDialogTitle disableTypography {...otherProps}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
};

export default DialogTitle;
