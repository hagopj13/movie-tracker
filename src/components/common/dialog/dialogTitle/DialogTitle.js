import React from 'react';
import type { Node } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

type Props = {
  children: Node,
  onClose: () => void,
};

const useStyles = makeStyles((theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(2),
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
}));

const DialogTitle = (props: Props) => {
  const { children, onClose, ...otherProps } = props;
  const classes = useStyles();

  return (
    <MuiDialogTitle disableTypography className={classes.root} {...otherProps}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
};

export default DialogTitle;
