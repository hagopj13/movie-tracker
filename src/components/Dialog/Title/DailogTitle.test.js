import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';

import DialogTitle from './DialogTitle';

describe('DialogTitle component', () => {
  let shallow;
  let title;
  let wrapper;

  const mockClose = jest.fn();

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    jest.clearAllMocks();
    title = 'Some title';
    const mockProps = {
      title,
      onClose: mockClose,
    };
    wrapper = shallow(<DialogTitle {...mockProps} />);
  });

  it('should render the DialogTitle component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the title', () => {
    expect(wrapper.find(Typography).text()).toBe(title);
  });

  it('should trigger the onClose prop when close icon is clicked', () => {
    wrapper.find(IconButton).simulate('click');
    expect(mockClose).toHaveBeenCalledTimes(1);
  });
});
