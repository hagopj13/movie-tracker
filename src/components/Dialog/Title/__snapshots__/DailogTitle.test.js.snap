// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogTitle component should render the DialogTitle component correctly 1`] = `
<WithStyles(ForwardRef(DialogTitle))
  disableTypography={true}
>
  <WithStyles(ForwardRef(Typography))
    variant="h6"
  >
    Some title
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(IconButton))
    aria-label="close"
    className="makeStyles-closeButton-1"
    onClick={[MockFunction]}
  >
    <CloseIcon />
  </WithStyles(ForwardRef(IconButton))>
</WithStyles(ForwardRef(DialogTitle))>
`;
